opm/autodiff/BlackoilModelEbos.hpp

```diff
@@ -626,8 +620,8 @@ namespace Opm {
                 p = std::max(p, 0.0);
 
                 // Saturation updates.
-                const double dsw = active_[Water] ? dx[cell_idx][Indices::waterSaturationIdx] : 0.0;
-                const double dxvar = active_[Gas] ? dx[cell_idx][Indices::compositionSwitchIdx] : 0.0;
+                const double dsw = FluidSystem::phaseIsActive(FluidSystem::waterPhaseIdx) ? dx[cell_idx][Indices::waterSaturationIdx] : 0.0;
+                const double dxvar = FluidSystem::phaseIsActive(FluidSystem::gasPhaseIdx) ? dx[cell_idx][Indices::compositionSwitchIdx] : 0.0;
```

Comment:
Similar here: I assume that the `Indices::compositionSwitchIdx` is only valid if we have a case with VAPOIL or DISGAS?

Reply:
If no vapoil and disgas the Indices::compositionSwitchIdx always refer to gas saturation.